### PR TITLE
Revert "feat(shuttle-axum) Make AxumService generic to be able to use axum::State with it (#924)"

### PR DESCRIPTION
**Pull request by jonaro00**
This reverts commit e6ade25efbca897a732ee3bb4f7de285b8e16d50.

## Description of change

An axum router with `<S != ()>` cannot be turned into a service. IT is only implemented for `<S = ()>` https://docs.rs/axum/latest/axum/struct.Router.html#impl-Router%3C(),+B%3E
The generic `<S>` means which state the router is missing. https://docs.rs/axum/latest/axum/routing/struct.Router.html#what-s-in-routers-means

## How has this been tested? (if applicable)

The example that was created for "verifying" #924, https://github.com/shuttle-hq/shuttle-examples/pull/52, did not verify the functionality of the generic (it's a good example tho). If you try setting S to anything, your code will now compile.
Basically, #924 was never fully tested.

### CODE REVIEW

**Comment by jonaro00** [comment]
@morlinbrot I'm curios what you think. Am I right or wrong?

**Review by orhun — APPROVED** [review]

**Review by iulianbarbu — APPROVED** [review]
